mixins/localize-mixin.js

```diff
@@ -20,13 +20,21 @@ export const LocalizeMixin = superclass => class extends superclass {
 			this.__languageChangeCallback = () => {
 				if (!this._hasResources()) return;
 				const possibleLanguages = this._generatePossibleLanguages();
-				this.constructor.getLocalizeResources(possibleLanguages)
-					.then((res) => {
-						if (!res) {
+				const localizeResources = this.constructor._getAllLocalizeResources(possibleLanguages);
+				const resourcesLoadedPromise = Promise.all(localizeResources);
+				resourcesLoadedPromise
+					.then((results) => {
+						if (results.length === 0) {
 							return;
 						}
-						this.__language = res.language;
-						this.__resources = res.resources;
+						const resources = {};
+						for (const res of results) {
+							const language = res.language;
+							for (const [key, value] of Object.entries(res.resources)) {
```

Comment:
Wondering if we can intercept the case where two mixins are specifying the same key and maybe throw? Otherwise I think depending on which order the promises resolve they'll get random results as to which one gets used.

Reply:
The order shouldn't be random. In `_getAllLocalizeResources` resources are loaded starting at the top of the inheritance chain working down so that the order of `localizeResources` has the same order. This allows mixins lower in the inheritance chain to override terms defined higher up in the inheritance chain.

Whether that's behavior we want or not, I'm open to convincing either way.

Reply:
Yeah I guess the order of the promises will be consistent... they may resolve out of order but you're right the array itself will be consistent.

I don't see this cropping up too often (it wasn't even possible until this PR!) so maybe we can ignore it. Mostly if we wanted to throw, now's our only opportunity.

Reply:
I do think allowing overriding might be slightly more intuitive.

The main issue I see with throwing an exception is that adding a term with a colliding key in a mixin could cause a component using that mixin to blow up even if the mixin itself works and passes all tests. 

Given that most of our UI repos don't have scheduled CI runs it could take a long time for it to be noticed if its in a repo that isn't having a ton of work done in it.

Reply:
Cool sounds good 👍